src/build_log.h

```diff
@@ -15,6 +15,16 @@
 #ifndef NINJA_BUILD_LOG_H_
 #define NINJA_BUILD_LOG_H_
 
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+#define NINJA_NULLPTR nullptr
+#else
+#define NINJA_NULLPTR NULL
```

Comment:
There's really no value to this typedef. NULL converts to nullptr where appropriate.

Better to just wait for C++11 to be merged, and then switch to nullptr wholesale.

Reply:
@jonesmz 
Thank you for your reviews :)

I tried to use `NULL`, but it seems that `NULL` cannot assign to `std::shared_ptr`.

```console
error: no viable conversion from 'long' to 'BuildLog::LogEntryPtr' (aka 'shared_ptr<BuildLog::LogEntry>')
  BuildLog::LogEntryPtr entry = NULL;
                        ^       ~~~~
```

WDYT?

Reply:
In those situations, you can wrap those lines with #ifdef, and use nullptr directly.